proto/regen/ecocredit/v1beta1/state.proto

```diff
@@ -0,0 +1,213 @@
+        primary_key: {fields: "project_id"}
+    };
+
+    // project_id is the unique ID of the project.
+    string project_id = 1;
+
+    // class_id is the unique ID of credit class for this project.
+    string class_id = 2;
+
+    // issuer is the issuer of the credit batches for this project.
+    string issuer = 3;
+
+    // project_location is the location of the project.
+    // Full documentation can be found in MsgCreateProject.project_location.
+    string project_location = 4;
+
+    // metadata is any arbitrary metadata attached to the project.
+    bytes metadata = 5;
+}
+
+// BatchInfo represents the high-level on-chain information for a credit batch.
+message BatchInfo {
+
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 3,
+        primary_key: {fields: "batch_denom"}
+        index: {id: 1 fields: "project_id"}
+        index: {id: 2 fields: "start_date"}
+    };
+
+    // project_id is the unique ID of the project this batch belongs to.
+    string project_id = 1;
+
+    // batch_denom is the unique ID of credit batch.
+    string batch_denom = 2;
+
+    // total_amount is the total number of active credits in the credit batch.
+    // Some of the issued credits may be cancelled and will be removed from
+    // total_amount and tracked in amount_cancelled. total_amount and
+    // amount_cancelled will always sum to the original amount of credits that
+    // were issued.
+    string total_amount = 4;
+
+    // metadata is any arbitrary metadata attached to the credit batch.
+    bytes metadata = 5;
+
+    // amount_cancelled is the number of credits in the batch that have been
+    // cancelled, effectively undoing there issuance. The sum of total_amount and
+    // amount_cancelled will always sum to the original amount of credits that
+    // were issued.
+    string amount_cancelled = 6;
+
+    // start_date is the beginning of the period during which this credit batch
+    // was quantified and verified.
+    google.protobuf.Timestamp start_date = 7 [ (gogoproto.stdtime) = true ];
+
+    // end_date is the end of the period during which this credit batch was
+    // quantified and verified.
+    google.protobuf.Timestamp end_date = 8 [ (gogoproto.stdtime) = true ];
+}
+
+// CreditType defines the measurement unit/precision of a certain credit type
+// (e.g. carbon, biodiversity...)
+message CreditType {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 4,
+        primary_key: {fields: "abbreviation"}
+    };
+
+    // the type of credit (e.g. carbon, biodiversity, etc)
+    string name = 1;
+
+    // abbreviation is a 1-3 character uppercase abbreviation of the CreditType
+    // name, used in batch denominations within the CreditType. It must be unique.
+    string abbreviation = 2;
+
+    // the measurement unit (e.g. kg, ton, etc)
+    string unit = 3;
+
+    // the decimal precision
+    uint32 precision = 4;
+}
+
+// SellOrder represents the information for a sell order.
+message SellOrder {
```

Comment:
Let's move orders into a separate proto package

Reply:
i separated it like:

`ecocredit/v1beta1/marketplace/state.proto`


is that how we want it?

Reply:
Yep 👍